astropy/units/quantity.py

```diff
@@ -1410,19 +1411,28 @@ def __format__(self, format_spec):
         as specifiers for the number.
 
         If the format specifier correctly applies itself to the value,
-        then it is used to format only the value. If it cannot be
-        applied to the value, then it is applied to the whole string.
+        then it is used to format only the value. If ``format_spec`` is a
+        known unit format name then it is only applied to the unit. In
+        other cases it is applied to the whole string.
 
         """
         try:
             value = format(self.value, format_spec)
             full_format_spec = "s"
+            unit = self._unitstr
         except ValueError:
             value = self.value
-            full_format_spec = format_spec
-
-        return format(f"{value}{self._unitstr:s}",
-                      full_format_spec)
+            try:
+                units_get_format(format_spec)
```

Comment:
Is this line just for the try: except? the returned formatter isn't used.

Reply:
My idea was to check if `format_spec` is a valid unit format by seeing if `units_get_format()` raises an exception, but I no longer think this is a good approach. I will replace these lines with something clearer.